File: Projekti/SQLAlchemy-Benchmark/repository/post_repository.py
from model.post import Post
from util.stopwatch import Stopwatch


class PostRepository:

    def __init__(self, session):
        self.session = session

    def update_posts_for_user(self, user_id, content):
        update_posts = self.session.query(Post).filter(
            Post.user_id == user_id).update({Post.content: content})
        self.session.commit()

File: Projekti/Hibernate-Benchmark/src/main/java/com/example/HibernateBenchmark/repository/PostRepository.java
package com.example.HibernateBenchmark.repository;

import com.example.HibernateBenchmark.model.Post;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.transaction.annotation.Transactional;

public interface PostRepository extends JpaRepository<Post, Integer> {

    @Modifying
    @Transactional
    @Query("UPDATE Post p SET p.content = :content WHERE p.user.userId = :userId")
    int updatePostContentForUser(@Param("content") String content, @Param("userId") Integer userId);
}
